src/rest/openapi.yaml

```diff
@@ -0,0 +1,455 @@
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Dataset"
+      responses:
+        "202":
+          description: Successfully created the active operational dataset.
+        "400":
+          description: Invalid request body.
+    put:
+      tags:
+        - node
+      summary: Update current active opertional dataset
+      description: |-
+        Updates the current active operational dataset on the current node. Only allowed if the Thread node is inactive. Create a
+        pending dataset if you would like to change the current active opertional dataset if the Thread node is active.
+      requestBody:
+        description: |-
+          Operational dataset object that will be stored as active operational dataset. Supports request body Content-Type `text/plain`
+          (dataset in TLV format as hex string) or `application/json` (dataset in JSOn format, JSON keys which are not set will be
+          initialized with default or random values).
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Dataset"
+          plain/text:
+            schema:
+              $ref: "#/components/schemas/TlvDataset"
+      responses:
+        "202":
+          description: Successfully updated the active operational dataset.
+        "400":
+          description: Invalid request body.
+        "404":
+          description: No active opertional dataset to update.
+  /node/dataset/pending:
+    get:
+      tags:
+        - node
+      summary: Get current pending operational dataset
+      responses:
+        "200":
+          description: Returns currently pending opertional dataset
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Dataset"
+            text/plain:
+              schema:
+                $ref: "#/components/schemas/TlvDataset"
+        "204":
+          description: No pending operational dataset
+    post:
+      tags:
+        - node
+      summary: Create new pending opertional dataset
+      description: |-
+        Creates new pending operational dataset on the current node. If the delay timer is specified, the new pending operational dataset
+        will become active after the time elapsed. The pending dataset will become empty.
+
+        Default parameters are choosen for parameters not set in the request body. By default, the delay timer is *not* set.
+
+        Note: To be considered as a valid active opertional dataset the ActiveTimestamp needs to be newer than the current active operational
+        dataset.
+      requestBody:
+        description: |-
+          Operational dataset object that will be stored as pending operational dataset. JSON keys which are not set will be initialized with
+          default or random values.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Dataset"
```

Comment:
Ditto, support datasetTlv?

Reply:
I decided against it as it would implement the same behavior as the `PUT` method. 

Since setting the DatasetTlv is idempotent, I felt `PUT` is more suitable. I guess its not really violating RESTful definitions if we would do the same in `POST`, but why have the same functionality in two methods?